### PR TITLE
Goto url

### PR DESCRIPTION
**Pull request by evasomething**

### CODE REVIEW

File: src/client/components/segment-bubble/segment-bubble.tsx [hunk 1]
```diff
@@ -102,6 +122,10 @@ export class SegmentBubble extends React.Component<SegmentBubbleProps, SegmentBu
           data-clipboard-text={label}
           onClick={this.closeMoreMenu.bind(this)}
         >{STRINGS.copyValue}</li>
+        <li
+          className="clipboard"
```
Comment (vogievetsky) [inline src/client/components/segment-bubble/segment-bubble.tsx]: no need for the clipboard class here. it is a special thing for the clipboard

File: src/client/components/segment-bubble/segment-bubble.tsx [hunk 2]
```diff
@@ -81,6 +83,24 @@ export class SegmentBubble extends React.Component<SegmentBubbleProps, SegmentBu
...
+      idx = nestIndex;
+    }
+    var url = urls[idx];
+
+    if (!url) url = defaultGoToUrl;
+    var replacedUrl = url.replace('%s', label);
+
+    window.open(
+      replacedUrl, '_blank'
+    );
```
Comment (vogievetsky) [inline src/client/components/segment-bubble/segment-bubble.tsx]: I think it is better to use an `<a>` element instead of doing it in jabascript, then people's shortcuts will work better

File: src/client/components/segment-bubble/segment-bubble.tsx [hunk 3]
```diff
@@ -81,13 +83,49 @@ export class SegmentBubble extends React.Component<SegmentBubbleProps, SegmentBu
...
 
+  getReplacedUrl(label: string) {
+    var { urls, datum } = this.props;
+    var idx = 0;
+    if (urls.length !== 1) {
+      const nestIndex = (datum['__nest'] as number) - 1;
+      idx = nestIndex;
+    }
+    var url = urls[idx];
+    return url.replace('%s', label);
```
Comment (vogievetsky) [inline src/client/components/segment-bubble/segment-bubble.tsx]: it is better to do `.replace(/%s/g, label)` it will replace all occurrences of `%s` which is usually more expected. Always be weary of having a string as the first argument in `replace()`, ask yourself is that what I really want?

File: src/client/components/segment-bubble/segment-bubble.tsx [hunk 4]
```diff
@@ -81,13 +83,49 @@ export class SegmentBubble extends React.Component<SegmentBubbleProps, SegmentBu
...
+        className="clipboard"
+        data-clipboard-text={label}
+        onClick={this.closeMoreMenu.bind(this)}
+      >{STRINGS.copyValue}</li>
+    ];
+    if (this.getHasUrl()) {
+      bubbleListItems = bubbleListItems.concat(
+        <li
+          key="goToUrl"
+        > <a href={this.getReplacedUrl(label)} onClick={this.closeMoreMenu.bind(this)} target="_blank" >
```
Comment (vogievetsky) [inline src/client/components/segment-bubble/segment-bubble.tsx]: OMG `> <` please make a new line

File: src/client/components/segment-bubble/segment-bubble.tsx [hunk 5]
```diff
@@ -81,13 +83,49 @@ export class SegmentBubble extends React.Component<SegmentBubbleProps, SegmentBu
...
+    var bubbleListItems = [
+      <li
+        key="copyValue"
+        className="clipboard"
+        data-clipboard-text={label}
+        onClick={this.closeMoreMenu.bind(this)}
+      >{STRINGS.copyValue}</li>
+    ];
+    if (this.getHasUrl()) {
+      bubbleListItems = bubbleListItems.concat(
```
Comment (vogievetsky) [inline src/client/components/segment-bubble/segment-bubble.tsx]: this could just as well be push as well as concat since it is all contained in one place (there is no danger of mutation)

File: src/common/models/dimension/dimension.ts [hunk 6]
```diff
@@ -89,7 +100,8 @@ export class Dimension implements Instance<DimensionValue, DimensionJS> {
       name: this.name,
       title: this.title,
       expression: this.expression.toJS(),
-      kind: this.kind
+      kind: this.kind,
+      url: this.url
```
Comment (vogievetsky) [inline src/common/models/dimension/dimension.ts]: oh, one more thing if there is something very optional like `url` in an immutable class that defaults to null. it might be better to not put it on the JS.
do `if (this.url) js.url = this.url;`

File: src/common/models/data-source/data-source.mocha.ts [hunk 7]
```diff
@@ -87,7 +90,8 @@ describe('DataSource', () => {
             },
             kind: 'string',
             name: 'articleName',
-            title: 'Article Name'
+            title: 'Article Name',
+            url: null
```
Comment (vogievetsky) [inline src/common/models/data-source/data-source.mocha.ts]: .. that way you do not need to change as many tests with `url: null`

**Comment by vogievetsky** [comment]
gr :8ball:
